src/Adapter/AttributeGroup/QueryHandler/GetAttributeGroupForEditingHandler.php

```diff
@@ -0,0 +1,85 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\AttributeGroup\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Exception\AttributeGroupException;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Query\GetAttributeGroupForEditing;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\QueryHandler\GetAttributeGroupForEditingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\QueryResult\EditableAttributeGroup;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupId;
+use PrestaShop\PrestaShop\Core\Domain\Product\AttributeGroup\Exception\AttributeGroupNotFoundException;
+use PrestaShopException;
+
+/**
+ * Handles query which gets attribute group for editing
+ */
+final class GetAttributeGroupForEditingHandler implements GetAttributeGroupForEditingHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetAttributeGroupForEditing $query): EditableAttributeGroup
+    {
+        $attributeGroupId = $query->getAttributeGroupId();
+        $attributeGroup = $this->getAttributeGroup($attributeGroupId);
+
+        return new EditableAttributeGroup(
+            $attributeGroupId,
+            $attributeGroup->name,
+            $attributeGroup->public_name,
+            $attributeGroup->group_type,
+            $attributeGroup->getAssociatedShops()
+        );
+    }
+
+    /**
+     * Gets legacy Attribute group
+     *
+     * @param AttributeGroupId $attributeGroupId
+     *
+     * @return \AttributeGroup
+     *
+     * @throws AttributeGroupException
+     * @throws AttributeGroupNotFoundException
+     */
+    protected function getAttributeGroup(AttributeGroupId $attributeGroupId): \AttributeGroup
+    {
+        try {
+            $attributeGroup = new \AttributeGroup($attributeGroupId->getValue());
+        } catch (PrestaShopException $e) {
+            throw new AttributeGroupException('Failed to create new attribute group', 0, $e);
+        }
+
+        if ($attributeGroup->id !== $attributeGroupId->getValue()) {
+            throw new AttributeGroupNotFoundException(sprintf('Attribute group with id "%s" was not found.', $attributeGroupId->getValue()));
+        }
+
+        return $attributeGroup;
```

Comment:
inject PrestaShop\PrestaShop\Adapter\AttributeGroup\Repository\AttributeGroupRepository
and a new method if it doesn't yet exist `get(AttributeGroupId $attributeGroupId): AttributeGroup` and use it here instead